### PR TITLE
cleanup: refactor samples to C++ driver

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #3489

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3495)

### CODE REVIEW

File: google/cloud/bigtable/examples/bigtable_grpc_credentials.cc [hunk 1]
```diff
@@ -23,34 +22,20 @@ struct Usage {
...
-  argc--;
-  return result;
-}
-
 std::string command_usage;
 
-void PrintUsage(int, char* argv[], std::string const& msg) {
-  std::string const cmd = argv[0];
+void PrintUsage(std::string const& cmd, std::string const& msg) {
   auto last_slash = std::string(cmd).find_last_of('/');
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_grpc_credentials.cc]: Opt: The `std::string()` around `cmd` looks like a confusing remnant.

File: google/cloud/bigtable/examples/bigtable_grpc_credentials.cc [hunk 2]
```diff
@@ -152,18 +129,50 @@ void GCECredentials(int argc, char* argv[]) {
...
   }
   //! [test gce credentials]
-  (std::move(project_id), std::move(instance_id));
+  (argv[0], argv[1]);
+}
+
+void RunAll(std::vector<std::string> argv) {
+  if (!argv.empty()) throw Usage{"auto"};
+
+  auto const project_id =
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_grpc_credentials.cc]: Opt Nit: Might it be clearer to run the checking `for` loop first, and then make these just `*GetEnv()` calls?

Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_grpc_credentials.cc]: Done.

File: google/cloud/bigtable/examples/bigtable_grpc_credentials.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_grpc_credentials.cc]: Fixed.

**Pull request merged**